src/ember-app.js

```diff
@@ -287,77 +111,203 @@ class EmberApp {
       fastbootInfo: fastbootInfo
     });
 
-    let destroyAppInstanceTimer;
-    if (parseInt(destroyAppInstanceInMs, 10) > 0) {
-      // start a timer to destroy the appInstance forcefully in the given ms.
-      // This is a failure mechanism so that node process doesn't get wedged if the `visit` never completes.
-      destroyAppInstanceTimer = setTimeout(function() {
-        if (instance && !result.instanceDestroyed) {
-          result.instanceDestroyed = true;
-          result.error = new Error('App instance was forcefully destroyed in ' + destroyAppInstanceInMs + 'ms');
-          instance.destroy();
-        }
-      }, destroyAppInstanceInMs);
-    }
-
-    let instance;
-    return this.visitRoute(path, fastbootInfo, bootOptions, result)
-      .then(appInstance => {
-        instance = appInstance;
+    let destroy;
+    return this.app.boot()
```

Comment:
I tried this, but breaking up the chain resulted in less readable code, especially when you're trying to avoid object scope (`this`) references to `destroy` and `instance`.

We could simplify this a lot by dealing with the destruction timer differently using `vm`'s default timeout mechanism: https://nodejs.org/api/vm.html#vm_new_vm_script_code_options

I did not want to attempt that refactor in this PR though.